### PR TITLE
Introduce model generator component

### PR DESCRIPTION
**Pull request by doosuu**
Introduces the model generator component which allows auto-generation of a JSON based vspec file upon component initialization

### CODE REVIEW

File: model-generator/src/install_deps.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    return False
+
+
+def install_packages():
+    """Install all required Python packages for the model generator and VSpec download."""
+    script_path = get_script_path()
+
+    model_gen_version = "v0.3.0"
+    model_gen_repo = "https://github.com/eclipse-velocitas/vehicle-model-generator.git"
```
Comment (BjoernAtBosch) [inline model-generator/src/install_deps.py:45]: Don't like that we have dependencies defined here "somewhere" in the code. Maybe we can move this later to a more central place.

**Review by BjoernAtBosch — APPROVED** [review]
LGTM

**Review by BjoernAtBosch — CHANGES_REQUESTED** [review]
Sorry - test run missing

File: manifest.json [hunk 2]
```diff
@@ -37,6 +37,52 @@
...
+                {
+                    "id": "generate-model",
+                    "executable": "python",
+                    "args": ["./model-generator/src/generate_model.py"]
+                }
+            ],
+            "variables": [
+                {
+                    "name": "installModelGenerator",
+                    "type": "bool",
```
Comment (dennismeister93) [inline manifest.json:75]:
```suggestion
                    "type": "boolean",
```

File: manifest.json [hunk 3]
```diff
@@ -37,6 +37,52 @@
...
+                    "id": "generate-model",
+                    "executable": "python",
+                    "args": ["./model-generator/src/generate_model.py"]
+                }
+            ],
+            "variables": [
+                {
+                    "name": "installModelGenerator",
+                    "type": "boolean",
+                    "default": false,
```
Comment (dennismeister93) [inline manifest.json:76]:
```suggestion
```

File: model-generator/src/install_deps.py (hunk 1 above)
Comment (doosuu) [reply model-generator/src/install_deps.py]: Moved this configuration to the variables section of the `manifest.json` - now its possible to use the component defaults or overwrite it in a project's `.velocitas.json`

**Review by dennismeister93 — APPROVED** [review]
LGTM

**Review by BjoernAtBosch — APPROVED** [review]
LGTM
(Tested with CPP template)

**Pull request merged**